week2/homework/src/index.js

```diff
@@ -1,3 +1,82 @@
-'use strict';
+'use strict:';
 
-// TODO: Write the homework code in this file
+const fs = require('fs');
+const { promisify } = require('util');
+const dataToNewLine = data => `\n${data}`;
+
+const writeFile = promisify(fs.writeFile);
+const readfile = promisify(fs.readFile);
+
+const filePath = './ToDoList.json';
+
+const getAllToDos = () => {
+  return readfile(filePath, 'utf8')
+    .then(data => JSON.parse(data))
+    .catch(err => console.log(err));
+};
+
+const updateTodo = async (id, newTitle) => {
+  const { todos } = await getAllToDos();
+  const updatedTodos = todos.map(todo => {
+    if (todo.id === id) {
+      return {
+        id,
+        title: newTitle,
+      };
+    }
+    return todo;
+  });
+
+  const writeData = JSON.stringify({ todos: updatedTodos }, null, 2);
+  await writeFile(filePath, writeData);
+};
+
+const writeToDos = async newTitle => {
+  const { todos } = await getAllToDos();
+  const todo = { id: todos.length + 1, title: newTitle };
+  const newTodos = todos.concat(todo);
+  const writeData = JSON.stringify({ todos: newTodos }, null, 2);
+  await writeFile(filePath, writeData);
+};
+
+const deleteToDo = async id => {
+  const { todos } = await getAllToDos();
+  const deleteToDo = todos.filter(todo => todo.id !== id);
+
+  const writeData = JSON.stringify({ todos: deleteToDo }, null, 2);
+  await writeFile(filePath, writeData);
```

Comment:
these two lines seem to repeat in every function.

Reply:
Yes, it is to make JSON and write it every time.